### ISSUE TITLE
PicklingError in data loader when num_workers>0

### ISSUE THREAD

**Issue by arglog** [pyg-team/pytorch_geometric#3143]
## 🐛 Bug

PyG 2.0.0 Dataloader raises errors when `num_workers` is greater 0

## To Reproduce
Run the script below (notice that `num_workers=1`)
```python
from torch_geometric.datasets import TUDataset
from torch_geometric.loader import DataLoader

dataset = TUDataset(root='/tmp/ENZYMES', name='ENZYMES', use_node_attr=True)
loader = DataLoader(dataset, batch_size=32, shuffle=True, num_workers=1)

for batch in loader:
    print(batch)
    break
```

The above script will trigger the following error
```
Traceback (most recent call last):
  File "/mnt/software/anaconda3/lib/python3.7/multiprocessing/queues.py", line 236, in _feed
    obj = _ForkingPickler.dumps(obj)
  File "/mnt/software/anaconda3/lib/python3.7/multiprocessing/reduction.py", line 51, in dumps
    cls(buf, protocol).dump(obj)
_pickle.PicklingError: Can't pickle <class 'torch_geometric.data.batch.Batch'>: it's not the same object as torch_geometric.data.batch.Batch
```

## Expected behavior
See above

## Environment

* OS: Ubuntu 16.04
* Python version: 3.8
* PyTorch version: py3.7_cuda11.1_cudnn8.0.5_0
* CUDA/cuDNN version: CUDA 11.1, cuDNN 8.0.5
* GCC version: 5.4.1
* PyG version: 2.0.0

**Comment by rusty1s** [comment]
Yes, I'm aware of this. I'm really sorry :(
It is already fixed in master:
```
pip install git+https://github.com/pyg-team/pytorch_geometric.git
```

**Comment by rusty1s** [comment]
I just released `torch-geometric==2.0.1` that fixes this issue :)

**Comment by RasmusOrsoe** [comment]
`<class 'torch_geometric.data.batch.Batch'>: it's not the same object as torch_geometric.data.batch.Batch ` seems to appear also when used in `Queue` from `torch.multiprocessing.Manager()` (used to work just fine!)

I load batches into a Queue using the code below
```
from torch_geometric.loader import DataLoader
loader = DataLoader(data_list_train,batch_size = batch_size,drop_last=True)
loader_it = iter(loader)
for k in range(0,len(loader)):
    q.put(next(loader_it))
```

And when I retrieve a batch using `q.get()` the error appears.

torch-geometric                    2.0.1
torch                              1.9.1+cu111
Python 3.8.5
CentOS Linux 7

Before updating this worked fine with the following versions:
torch==1.7.1+cu110
torch-geometric==1.6.3
Python 3.8.5
CentOS Linux 7
